pkgs/by-name/ke/keycastr/package.nix

```diff
@@ -0,0 +1,36 @@
+{
+  stdenvNoCC,
+  lib,
+  fetchurl,
+  unzip,
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "keycastr";
+  version = "0.9.18";
+
+  src = fetchurl {
+    url = "https://github.com/keycastr/keycastr/releases/download/v${finalAttrs.version}/KeyCastr.app.zip";
+    hash = "sha256-q12c/W0yGoVL+wx+T/gaevx2P0Xwcv0a0FMv7bKfUnE=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,Applications}
+    cp -r KeyCastr.app $out/Applications/
```

Comment:
```suggestion
    install -Dm755 KeyCastr.app $out/Applications/
```

Reply:
Darwin apps are just folders, and `install` - as far as I'm aware - does not not perform recursive copy of folders

Reply:
Oh didn't know that, I thought it was something like appimage